Provide option details (#361)
magento/architecture: design-documents/graph-ql/coverage/add-items-to-cart-single-mutation.md

- Options are sets of predefined values that a buyer can add to cart/wishlist in addition to a product.
- The difference between wish list and cart payloads is that all fields in wishlist are optional.

### Options 

Each product may have options. Option can be of 2 types (see example below):
  - `selected_options` - predefined and selected by customer option. E.g. it can be customizable option: `color: green` or bundle option: `Memory: 24M, Warranty: 1y`:
``` 
    "selected_options" : [
         "Y3VzdG9tLW9wdGlvbi8yNC80Mg=="
     ],
```
  - `entered_options` - option entered by customer like: text field, image, etc.
``` 
    "entered_options": [
      {
        id: "Y3VzdG9tLW9wdGlvbi8zMQ==",
        value: "Vasia's awesome cup"
      }
    ]
```

We can consider "Selected Option" and "ID for Entered Option" as UUID. They meet the criteria:

> - Represent specific option. 

Reviewer: "Selected Option" represents option value, while "ID for Entered Option" represents option.

Author: done

- Must be unique across different options
- Returned from server
- Used by client as is

Selected options can be used for:
- Customizable options such as dropdwon, radiobutton, checkbox, etc
- Configugrable product
- Bundle Product
- Downloadable product
- Groupped product

Entered options:
- Customizable options such as text field, file, etc
- Gift Card (amount)
  

> ### Option implementation

Reviewer: ```suggestion
#### Option implementation
```

  
Product schema should be extended in order to provide option UUID.

> Option UUID is `base64` encoded string, that pack details for each option and in most cases is can be preseneted as 

Reviewer: ```suggestion
Option UUID is `base64` encoded string, that encodes details for each option and in most cases can be presented as 
```

> Option UUID is `base64` encoded string, that encodes details for each option and in most cases can be presented as 

Reviewer 2: this field will return value which is not a uuid, right? to avoid confusion can we use valueId or id name?

Author: Yes, currently it will not be real UUID. In future we can use this field as it should be

Reviewer 2: uuid is an implementation for ID field, please be more abstract in naming.

Author: Renamed to id_v2. Please, check
CC: @paliarush

`base64("<option-type>/<option-id>/<option-value-id>")`

> For example, for customizable drop-down option "Color(id = 1), with values Red (id=1, Green=2)" UUID for Color:Red will looks like `"Y3VzdG9tLW9wdGlvbi8xLzE=" => base64("custom-option/1/1")` 

Reviewer: ```suggestion
For example, for customizable drop-down option "Color(id = 1), with values Red(id = 1), Green(id = 2)" UUID for Color:Red will looks like `"Y3VzdG9tLW9wdGlvbi8xLzE=" => base64("custom-option/1/1")` 
```

> Here is a GQL query that shows where need to add new field "uuid: String!" to cover existing cases:

Reviewer: ```suggestion
Here is a GQL query that shows how to add a new field "uuid: String!" to cover existing cases:
```
